cmd/oras/internal/errors/errors.go

```diff
@@ -60,6 +65,51 @@ func CheckArgs(checker func(args []string) (bool, string), Usage string) cobra.P
 	}
 }
 
+// Handler handles error during cmd execution.
+type Handler interface {
+	// Handle handles error during cmd execution.
+	// If returned processor is nil, error requires no further processing.
+	Handle(err error, cmd *cobra.Command) (Processor, error)
+}
+
+// Processor processes error.
+type Processor interface {
+	Process(err error, callPath string) *Error
+}
+
+// Command returns an error-handled cobra command.
+func Command(cmd *cobra.Command, handler Handler) *cobra.Command {
+	runE := cmd.RunE
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		err := runE(cmd, args)
+		if err == nil {
+			return nil
+		}
+		processor, err := handler.Handle(err, cmd)
+		if processor == nil {
+			return err
+		}
+		return processor.Process(err, cmd.CommandPath())
```

Comment:
The design of `Handler` and `Processor` seems can be simplified.

Here's the analysis:
1. The `err` is returned by `handler.Handle()`. That implies `Handle` has the information of `err`.
2. Similarly, `Handle` has the information of `processor`.

Combine 1 and 2, we can simplify the above code as
```go
processor := handler.Handle(err, cmd)
processor.Process(cmd.CommandPath())
```
or simply
```go
return handler.Handle(err, cmd).Process(cmd.CommandPath())
```
which can be further simplified as
```go
type ErrorHandler interface {
	HandleError(cmd *cobra.Command, err error) error
}
```
and
```go
return handler.HandleError(cmd, err)
```

Reply:
Tried to use `Hande` and `Process` to decouple the logic of setting output prefix, scrubbing error info and give recommendation for know errors. I have merged the two interfaces.